### ISSUE TITLE
Error::NoPassword is useless ?

### ISSUE THREAD

**Issue by GaetanLepage** [dnaka91/obws#37]
When connecting to a server that requires password authentification, with
```rust
let pw: Option<String> = None;
let client_res = Client::connect("localhost", 4455, pw).await();
...
```
it fails with
```
Handshake(ConnectionClosed(Some(CloseDetails { code: Library(4009), reason: "Your payload's data is missing an `authentication` string, however authentication is required." })))
```

I would have expected a `Error::NoPassword` error for this case.
Furthermore, I found no references to this error variant in the `obws` code base. Is it indeed unused ?

**Comment by dnaka91** [comment]
It is indeed not used (anymore).

The authentication mechanism changed a lot with the obs-websocket API **v5** which removed the place where it was used. In the past connecting and autehtnicating were to separate steps obws could check the connect response for required authentication.

Now with the new API the authentication has to happen upfront during the connect, which made this error obsolete.

Long story short it's a relic of the old API and I'll remove the error variant with the next version as it's not used anymore and there's no way to make it useful anymore (the API doesn't give any specific detail about why the authentication failed).

**Comment by GaetanLepage** [comment]
Thank you for confirming this.

> and there's no way to make it useful anymore (the API doesn't give any specific detail about why the authentication failed).

I guess there would still be a (flaky) way to detect this: If the error is an Handshake-ConnectionClosed with `"Your payload's data is missing an `authentication` string, however authentication is required." as the `reason`. But I guess that it is better to leave it as is.

**Comment by dnaka91** [comment]
Yeah I don't want to rely on checking the error message. Any time they decide to change the wording it would break things, and I don't believe they reserve pushing those changes for the next major version release.

Would be great if they provide some extra details, so I could translate the response to specific error variant.

**Comment by GaetanLepage** [comment]
Sure, this makes sense ! Thanks for your answer.
Feel free to close this issue whenever you want.

**Comment by dnaka91** [comment]
I just released a new version (v0.12) which removes this error variant besides several other changes.

The changelog now features a short description of how to get something as close as possible to this variant. Basically when you get a `Error::Handshake` you get extract the websocket close code and compare it against the [`WebSocketCloseCode::AuthenticationFailed`](https://docs.rs/obws/latest/obws/responses/enum.WebSocketCloseCode.html#variant.AuthenticationFailed).
